File: Tccp.Algorithm/MovingTotal/MovingTotal.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Tccp.Algorithm
{
    public class MovingTotal
    {
        List<int> items = new List<int>();
        public void Append(int[] list)
        {
            this.items.AddRange(list);
        }

        public bool Contains(int total)
        {
            int[] array = this.items.ToArray();
            var size = array.Length - 1;
            var indexToMove = 0;
            var sum = 0;
            while (sum <= total)
            {
                if (sum == total) return true;

                if (indexToMove > size || indexToMove + 1 > size || indexToMove + 2 > size) return false;

                var number0 = array[indexToMove];
                var number1 = array[indexToMove + 1];
                var number2 = array[indexToMove + 2];
                sum = number0 + number1 + number2;

                if (sum == total) return true;
                indexToMove += 1;
            }
            return false;
        }
    }
}

File: Python/algorithms/moving_total.py
from __future__ import annotations


class MovingTotal:
    def __init__(self) -> None:
        self.items: list[int] = []

    def append(self, values: list[int]) -> None:
        self.items.extend(values)

    def contains(self, total: int) -> bool:
        if len(self.items) < 3:
            return False

        for i in range(len(self.items) - 2):
            if self.items[i] + self.items[i + 1] + self.items[i + 2] == total:
                return True
        return False
